packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -347,6 +347,7 @@ function TableVirtualizer({layout, collection, focusedKey, renderView, renderWra
         {state.visibleViews[0]}
       </div>
       <ScrollView
+        sizeToFit="height"
```

Comment:
Not entirely sure about this fix, trying to puzzle out what exactly the logic in scrollView is doing (deciding what the visible rect height would be, e.g. the "viewport" rectangle height displaying the items)?

Also the `scrollOffset` calculation in onVisibleRectChange confuses me a bit, I figured it would be something more like this:
```
      // When to load more items? Well it would be when the scroll position (SP) + the "item viewport" height (VH) is equal or greater than the total content height (CH).
      //  SP + VH >= CH   -->  becomes SP >= CH -VH --> rect.y >= state.virtualizer.contentSize.height - rect.height
      let scrollOffset = state.virtualizer.contentSize.height - rect.height;
      if (rect.y >= scrollOffset) {
        collection.body.props.onLoadMore();
      }
``` 
Changing it to that make it properly load one page of extra results rather than two pages of extra result but not sure if I'm misunderstanding the current logic

Reply:
This will cause the scrollview's height to match its content. This means that if you have 1,000 rows, all 1,000 will be rendered because the scroll view's height will be sized to fit all of them. I don't think we really want that.

Reply:
It takes the minimum between dom element's height and the total content height though right? So we would get bounded by the Table's height/the Table container's height (if Table height isn't explicitly defined)?

Reply:
Won't this reduce the size of the table itself though? i.e. if there are not enough rows to fill the viewport, the viewport will be reduced in height so that it matches the number of rows. I don't think that's expected behavior for a table - it should be the height the user sets.

Reply:
As it is right now, it doesn't reduce the size of the table. The height that the user sets is applied to the [div wrapping the TableHeader and ScrollView](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/table/src/Table.tsx#L252). The [ScrollView itself has `flex: 1`](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/table/src/Table.tsx#L352) applied to it so it grows to match the remainder of the height left by the TableHeader. The internal height state that ScrollView tracks doesn't seem to affect the component's height (at least from what I can tell).

The way I understood the logic of "when to loadMore" when `sizeToFit: "height"` is applied was as follows:

if total content height (CH) is less than viewport height (VP) (aka not enough rows to fill viewport):
- ScrollView updates its internal height tracker to be the min between the content height and the viewport height. Since content height is the lesser value, it sets it equal to content height 
- this triggers `onVisibleRectChange` with the new height (CH). We now know that we have to load more items because the viewport rectangle's `y` position (0) is greater than the total content height minus the new height times 2 (CH - 2*VP --> CH -2CH)
- future scroll events/loadMore events will adjust ScrollView's internal height tracker to be equal to the original viewport height (aka the outer div minus the table header) since total content height becomes larger than the viewport height

if total content height (CH) is greater than viewport height (VP) (aka more than enough rows to fill viewport):
- ScrollView updates its internal height tracker to be the min between the content height and the viewport height. Since viewport height is the lesser value, it sets it equal to the viewport height (aka the outer div minus the table header height)
- this triggers `onVisibleRectChange` with the new height (VP). We can calculate if we need to load more items if the viewport rectangle's `y` position (0) is less than the total content height minus the viewport height times 2. This formula feels a bit off to me as I mentioned previously since content heights that are 2X or less the viewport height will then cause a loadMore to be triggered even though the content is more than enough to fill the view port. 
0 > CH - 2*VP --> if CH <= 2*VP it will call loadMore